frontend/server/utils.ts

```diff
@@ -65,6 +66,32 @@ export function loadJSON<T>(filepath?: string, defaultValue?: T): T | undefined
   }
 }
 
+/**
+ * prune remainingMaybePrunePath to relativePath, then concat with rootPath.
+ * for example:
+ * console.log(pruneAndConcatPath('/data', 'a/b/c', undefined))
+ * console.log(pruneAndConcatPath('/data', 'a/b/c', a))
+ * console.log(pruneAndConcatPath('/data', 'a/b/c', a/b))
+ * console.log(pruneAndConcatPath('/data', 'a/b/c', other))
+ *
+ * result is:
+ * /data/a/b/c
+ * /data/b/c
+ * /data/c
+ * /data/a/b/c
+ */
+export function pruneAndConcatPath(
+  rootFixedPath: string,
+  remainingMaybePrunePath: string,
+  prefixPrunePath: string | undefined,
+): string {
```

Comment:
the function signature is very hard to understand to me, although I could guess its meaning from our previous discussion.

let me think about it and give you a suggestion later.